rpcs3/Emu/Cell/Modules/cellGame.cpp

```diff
@@ -786,14 +786,72 @@ error_code cellGameCreateGameData(vm::ptr<CellGameSetInitParams> init, vm::ptr<c
 
 error_code cellGameDeleteGameData(vm::cptr<char> dirName)
 {
-	cellGame.todo("cellGameDeleteGameData(dirName=%s)", dirName);
+	cellGame.warning("cellGameDeleteGameData(dirName=%s)", dirName);
 
 	if (!dirName)
 	{
 		return CELL_GAME_ERROR_PARAM;
 	}
 
-	return CELL_OK;
+	const std::string name = dirName.get_ptr();
+	const std::string dir = vfs::get("/dev_hdd0/game/"s + name);
+
+	const auto prm = g_fxo->get<content_permission>();
+
+	auto remove_gd = [&]() -> error_code
+	{
+		psf::registry sfo = psf::load_object(fs::file(dir + "/PARAM.SFO"));
```

Comment:
The flow of this function is completely broken.
First you create an object based on a file that you don't know exists.
Then you check if the file exists.
Then you try to delete everything in the folder, but only check if the dir even exists after the deletion fails.

Reply:
Ok, it is not broken but reordered so there wont be an fs::file object open while remove_all executes (can result in errors etc). I can move NOTFOUND check before remove_all so it will be less confusing.

Reply:
You can just put it in scope though

Reply:
psf::load_object explicitly checks file existense, seems redundent to check ourselfs it and not to use this feature. Adding sa scope will just result in uglier code, not to mention other cellGane function follow this pattern as well for these reasons.